debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -98,6 +98,27 @@
      * @return the connection factory
      */
     public static ConnectionFactory patternBasedFactory(String urlPattern, Field... variables) {
+        return patternBasedFactory(urlPattern, JdbcConnection.class.getClassLoader(), variables);
+    }    
+    
+    /**
+     * Create a {@link ConnectionFactory} that replaces variables in the supplied URL pattern. Variables include:
+     * <ul>
+     * <li><code>${hostname}</code></li>
+     * <li><code>${port}</code></li>
+     * <li><code>${dbname}</code></li>
+     * <li><code>${username}</code></li>
+     * <li><code>${password}</code></li>
+     * </ul>
+     * 
+     * @param urlPattern the URL pattern string; may not be null
+     * @param variables any custom or overridden configuration variables
+     * @param classloader when configuration has jdbc.driver specified, the driver will be loaded from this 
+     *        class loader if one is provided. When null DriverManager will resolve the driver based on urlPattern 
```

Comment:
This isn't correct, is it? When the _driver class name_ is null, the DriverManager will resolve the driver using the URL patter. Otherwise, the this method will attempt to load the driver class (optionally using the `classloader` if one is provided), create a new instance of the Driver with the URL and properties, and then attempt to connect.

The JavaDoc for the method needs to explain what this is doing.